xija/component/heat.py

```diff
@@ -869,18 +866,60 @@ def plot_solar_heat__pitch(self, fig, ax):
                 lines[i].set_data(self.P_pitches, P_vals[instr_name])
                 # lines[i * 2 + 1].set_data(self.P_pitches, P_vals[instr_name], '-b')
         else:
-            for instr_name, color in zip(self.instr_names, colors):
+            for i, instr_name in enumerate(self.instr_names):
+                color = colors[i]
                 ax.plot(self.P_pitches, P_vals[instr_name], 'o-{}'.format(color), markersize=5,
                         label=instr_name)
             ax.set_title('{} solar heat input'.format(self.node.name))
             ax.set_xlim(40, 180)
             ax.grid()
             ax.legend(loc='best')
 
+    def __str__(self):
+        return 'simz_solarheat__{0}'.format(self.node)
+
+
+class AcisPsmcSolarHeat(SimZDepSolarHeat):
+    """Solar heating of PSMC box.  This is dependent on SIM-Z"""
+    def __init__(self, model, node, pitch_comp='pitch', simz_comp='sim_z',
+                 dh_heater_comp='dh_heater', P_pitches=None,
+                 P_vals=None, dPs=None, var_func='linear',
+                 tau=1732.0, ampl=0.05, epoch='2013:001:12:00:00', dh_heater=0.05):
+        simz_lims = [[-400000.0, -85000.0],  # HRC-S
+                     [-85000.0, 0.0],        # HRC-I
+                     [0.0, 83000.0],         # ACIS-S
+                     [83000.0, 400000.0]]    # ACIS-I
+        instr_names = ['hrcs', 'hrci', 'aciss', 'acisi']
```

Comment:
What about making these class attributes (and maybe immutable tuples as well). Then the subclasses have only changes to the class attributes and the `__str__` but no `__init__`. The base class `SimZDepSolarHeat` would have those attributes set to `None` which means that this is not directly usable as a model component.

Otherwise this looks great at first glance, definitely a good idea.

Reply:
I agree